Fundamentals/Setup/Server-Setup/Load-Balancing/index-v9.md

```diff
@@ -0,0 +1,150 @@
+by default would mean the "umbracoApplicationUrl" is "f02.mysite.local". In any case the elected "Scheduling server" must be able to resolve this address.
+
+In many scenarios this is fine, but in case this is not adequate there's a few of options you can use:
+
+* __Recommended__: [set your front-end(s) (non-admin server) to be explicit replica servers](flexible-advanced-v9.md#explicit-master-scheduling-server) by creating a custom `IServerRegistrar`, this means the front end servers will never be used as the master scheduler
+* Set the `umbracoApplicationUrl` property in the [Web.Routing section of /Config/umbracoSettings.config](../../../../Reference/Config/umbracoSettings/index.md)
+
+## Common load balancing setup information
+
+_The below section applies to all ASP.NET load balancing configurations._
+
+## Server Configuration
+
+This section describes the various configuration options depending on your hosting setup:
+
+1. [Azure Web Apps](file-system-replication-v9.md#mixture-of-standalone--synchronised) - _You use cloud based auto-scaling appliances like [Microsoft's Azure Web Apps](https://azure.microsoft.com/en-us/services/app-service/web/)_
+2. [File Replication](file-system-replication-v9.md#synchronised-file-system) - _Each server hosts copies of the load balanced website files and a file replication service is running to ensure that all files on all servers are up to date_
+3. [Centralized file share](file-system-replication-v9.md#synchronised-file-system) - _The load balanced website files are located on a centralized file share (SAN/NAS/Clustered File Server/Network Share)_
+
+[Full documentation is available here](file-system-replication-v9.md)
+
+### Data Protection
+The replacement for Machine Keys in ASP.NET Core are called Data Protection.
+You will need to setup data protection to the same keys on all servers,
+without this you will end up with view state errors, validation errors and encryption/decryption errors since each server will have its own generated key.
+
+ASP.NET Core supports multiple ways to share keys. Use the [official docs](https://docs.microsoft.com/en-us/aspnet/core/security/data-protection/configuration/overview) to find a description that fits your setup the best.
+
+
+### Session State and Distributed Cache
+
+* If you use Session in your application, or are using the default TempDataProvider in MVC (which uses Session) then you will need to configure your application to use the `DistributedSqlServerCache` or an alternative provider (see [https://docs.microsoft.com/en-us/aspnet/core/performance/caching/distributed](https://msdn.microsoft.com/en-us/library/aa478952.aspx) for more details).
```

Comment:
@bergmania does v9 use TempData or ViewData in any of the front end methods, forms, surface controllers etc? If so that would require a distributed cache configuring when load balancing?

Reply:
As far as I know CMS do not use TempData for the frontend but Umbraco Forms does. I would always set up the distributed cache when configuring when load balancing.

Reply:
@bergmania cool, I think in v8 the BeginUmbracoForm and associated methods used TempData which was the reason session state was required

Reply:
You are right.. Seems like we use it in a couple of places
- All the member snippets use TempData
- The `RedirectToUmbracoUrlResult` and `RedirectToUmbracoPageResult` implements `IKeepTempDataResult`.

Guess you should always set up a distributed cache when using load balancing..